clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -22,11 +22,11 @@
     private final String topic;
     private final int partition;
     private final long offset;
-    private final long timestamp;
+    private final Long timestamp;
     private final TimestampType timestampType;
-    private final long checksum;
-    private final int serializedKeySize;
-    private final int serializedValueSize;
+    private final Long checksum;
+    private final Integer serializedKeySize;
+    private final Integer serializedValueSize;
```

Comment:
That is true. My reasoning was that these fields would never be null except for the compatibility case. And I was thinking that we could deprecate the old constructor. The main benefit is that users of these methods (the vast majority of use-cases) would not have to worry about null. But looking at it more carefully, users actually do have to handle it. See for example the following documentation:

```The size of the serialized, uncompressed key in bytes. If key is null, the returned size is -1.```

And `ProducerRecord` actually has a nullable timestamp. So, maybe this is indeed wrong and should be as you suggest.